### PR TITLE
New Location and MiniProfileCard components

### LINKED ISSUE #88: New shorter Profile Card version

### PR DESCRIPTION
**Pull request by maxme**
Closes #88

Screenshot_20240408_082511

### Description

New:
- Mini profile component (added to the demo app)
- Location component (which is just an ExpandableText using the profile location)

### Testing Steps

- Run the demo app
- Check the Profile tab
- Make sure the card is matching the design

### CODE REVIEW

File: gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.gravatar.ui.components.atomic.ViewProfileButton
+
+@Composable
+fun MiniProfileCard(profile: UserProfile, modifier: Modifier = Modifier) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.Top,
+    ) {
+        Row {
```
Comment (hamorillo) [inline gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt]: ⚠️  We don't need the column or the horizontal/vertical alignments.

```suggestion
    Row(modifier = modifier) {
```

File: gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            Avatar(
+                profile = profile,
+                size = 72.dp,
+                modifier = Modifier.clip(CircleShape),
+            )
+            Column(modifier = Modifier.padding(14.dp, 0.dp, 0.dp, 0.dp)) {
```
Comment (hamorillo) [inline gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt]: ⛏
```suggestion
            Column(modifier = Modifier.padding(start = 14.dp)) {
```

Should we also add the padding at the end?

```suggestion
            Column(modifier = Modifier.padding(horizontal = 14.dp)) {
```

File: demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt [hunk 3]
```diff
@@ -210,6 +211,14 @@ private fun ProfileTab(modifier: Modifier = Modifier, onError: (String?, Throwab
                         .fillMaxWidth()
                         .padding(24.dp, 24.dp, 24.dp, 24.dp),
                 )
+                Spacer(modifier = Modifier.height(16.dp))
+                MiniProfileCard(
+                    profiles.entry.first(),
+                    Modifier
+                        .background(MaterialTheme.colorScheme.surfaceContainer)
+                        .fillMaxWidth()
+                        .padding(24.dp, 24.dp, 24.dp, 24.dp),
```
Comment (hamorillo) [inline demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt]: ⛏
```suggestion
                        .padding(24.dp),
```

**Comment by hamorillo** [comment]
Looks nice! I left some minor comments, the more interesting could be [this one](https://github.com/Automattic/Gravatar-SDK-Android/pull/93#discussion_r1555497639).

File: gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt (hunk 2 above)
Comment (maxme) [reply gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt]: > Should we also add the padding at the end?

I'd keep the start padding only, it seems easier to manage in a card like in there:

Screenshot 2024-04-08 at 14 56 53



(24dp padding start and end on the card)

File: gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt (hunk 1 above)
Comment (maxme) [reply gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt]: Updated in ef36d1370ffed46828e611176de890930d7497d1

File: demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt (hunk 3 above)
Comment (maxme) [reply demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt]: Updated in ef36d1370ffed46828e611176de890930d7497d1

**Comment by maxme** [comment]
I updated the code according to your comments and I added a "maxLines" option to the location component, also changed the default to 1 line as it seems more appropriate for a location only.

**Review by hamorillo — APPROVED** [review]
:shipit:

**Pull request merged**